src/main/utils/shell-env/compute-unix-shell-environment.injectable.ts

```diff
@@ -7,96 +7,149 @@ import { getInjectable } from "@ogre-tools/injectable";
 import getBasenameOfPathInjectable from "../../../common/path/get-basename.injectable";
 import spawnInjectable from "../../child-process/spawn.injectable";
 import randomUUIDInjectable from "../../crypto/random-uuid.injectable";
+import loggerInjectable from "../../../common/logger.injectable";
+import processExecPathInjectable from "./execPath.injectable";
+import processEnvInjectable from "./env.injectable";
+import { object } from "../../../common/utils";
+import type { AsyncResult } from "../../../common/utils/async-result";
 
 export interface UnixShellEnvOptions {
   signal: AbortSignal;
 }
 
-export type ComputeUnixShellEnvironment = (shell: string, opts: UnixShellEnvOptions) => Promise<EnvironmentVariables>;
+export type ComputeUnixShellEnvironment = (shell: string, opts: UnixShellEnvOptions) => Promise<AsyncResult<EnvironmentVariables, string>>;
+
+const getResetProcessEnv = (src: Partial<Record<string, string>>, overrides: Partial<Record<string, string>>): {
```

Comment:
Short description for the function would make this easier to read.

Reply:
Fair, I can add that quickly